<launch>
    <!-- <arg name="name" default="localbot"/> -->
    <arg name="name" default="localbot"/>
    <arg name="color" default="Grey"/>
    <arg name="x_pos" default="0.0"/>
    <arg name="y_pos" default="0.0"/>
    <arg name="z_pos" default="0.5"/>
    <arg name="R_pos" default="0"/>
    <arg name="P_pos" default="0"/>
    <arg name="Y_pos" default="0"/>
    <arg name="rviz" default="true"/>

    <!--Load robot xacro into robot_description ros parameter.-->
    <param name="robot_description" command="$(find xacro)/xacro '$(find synfeal_bringup)/camera_description/urdf/localbot.urdf.xacro'
                                                 name:=$(arg name) color:=$(arg color)"/>

    <!-- <param name="robot_description" command="$(find xacro)/xacro '$(find synfeal_bringup)/camera_description/urdf/axes.urdf.xacro'
                                                 name:=$(arg name) color:=$(arg color)"/> -->


    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>

    <include file="$(find synfeal_bringup)/launch/secondary/spawn.launch">
        <arg name="name" value="$(arg name)"/>
        <arg name="x_pos" default="$(arg x_pos)"/>
        <arg name="y_pos" default="$(arg y_pos)"/>
        <arg name="z_pos" default="$(arg z_pos)"/>
        <arg name="R_pos" default="$(arg R_pos)"/>
        <arg name="P_pos" default="$(arg P_pos)"/>
        <arg name="Y_pos" default="$(arg Y_pos)"/>
    </include>

    <group if="$(eval arg('rviz') == true)">
        <include file="$(find synfeal_bringup)/launch/secondary/visualize.launch"/>
    </group>

    <!-- throttle because otherwise there are redundant tfs-->
    <node name="throttle" pkg="topic_tools" type="throttle" args= "messages /gazebo/model_states 10 /gazebo/model_states_throttle" output='screen' />
    <!-- Sleep 1 is to give time for the spawn to be finished-->
    <node name="model_states_to_tf" pkg="synfeal_bringup" type="model_states_to_tf.py" launch-prefix="bash -c 'sleep 1; $0 $@' " output = 'screen' /> 
    
    

</launch>